cluster-bootstrap/src/test/scala/akka/management/cluster/bootstrap/contactpoint/ClusterBootstrapDiscoveryBackoffIntegrationSpec.scala

```diff
@@ -0,0 +1,164 @@
+
+    val baseTime = System.currentTimeMillis()
+    case class DiscoveryRequest(time: Long, attempt: Int, res: Future[_]) {
+      override def toString = s"DiscoveryRequest(${(time - baseTime).millis}, $attempt, $res)"
+    }
+    val resolveProbe = TestProbe()(systemA)
+
+    // prepare the "mock DNS" - this resolves to unreachable addresses for the first three
+    // times it is called, thus testing that discovery is called multiple times and
+    // that formation will eventually succeed once discovery returns reachable addresses
+
+    @volatile var called = 0
+    @volatile var call2Timestamp = 0L
+    @volatile var call3Timestamp = 0L
+    val name = "system.svc.cluster.local"
+
+    MockDiscovery.set(name, { () =>
+      called += 1
+
+      if (called == 2)
+        call2Timestamp = System.nanoTime()
+      else if (called == 3)
+        call3Timestamp = System.nanoTime()
+
+      val res =
+        if (called < 4)
+          Future.failed(new Exception("Boom! Discovery failed, was rate limited for example..."))
+        else
+          Future.successful(Resolved(name,
+              List(
+                ResolvedTarget(clusterA.selfAddress.host.get, contactPointPorts.get("A")),
+                ResolvedTarget(clusterB.selfAddress.host.get, contactPointPorts.get("B"))
+              )))
+
+      resolveProbe.ref ! DiscoveryRequest(System.currentTimeMillis(), called, res)
+      res
+    })
+
+    "start listening with the http contact-points on 3 systems" in {
+      def start(system: ActorSystem, contactPointPort: Int) = {
+        import system.dispatcher
+        implicit val sys = system
+        implicit val mat = ActorMaterializer()(system)
+
+        val bootstrap = ClusterBootstrap(system)
+        val routes = new HttpClusterBootstrapRoutes(bootstrap.settings).routes
+        bootstrap.setSelfContactPoint(s"http://127.0.0.1:$contactPointPort")
+        Http().bindAndHandle(RouteResult.route2HandlerFlow(routes), "127.0.0.1", contactPointPort)
+      }
+
+      start(systemA, contactPointPorts("A"))
+      start(systemB, contactPointPorts("B"))
+    }
+
+    "poll discovery in exponentially increasing backoffs until eventually joining the returned nodes" in {
+      bootstrapA.discovery.getClass should ===(classOf[MockDiscovery])
+
+      bootstrapA.start()
+      // bootstrapB.start() // deliberately not discovering from this node, however its management port is available
+
+      val pA = TestProbe()(systemA)
+      clusterA.subscribe(pA.ref, classOf[MemberUp])
+
+      pA.expectMsgType[CurrentClusterState]
+      val up1 = pA.expectMsgType[MemberUp](45.seconds)
+      info("" + up1)
+
+      called shouldBe >=(5)
+
+      val durationBetweenCall2And3 = (call3Timestamp - call2Timestamp).nanos
+      info(s"duration between call 2 and 3 ${durationBetweenCall2And3.toMillis} ms")
+      durationBetweenCall2And3 shouldBe >=(ClusterBootstrap(systemA).settings.contactPointDiscovery.interval * 2)
```

Comment:
@ktoso I added this test, so that it actually fails without the backoff

reduced the interval in test to 500ms

also squashed, should be ready for merge now

Reply:
Ah nice! Yeah that's nice and simpler than I was trying to do here... Thanks a ton :)
Let's merge once validates then